components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! This module contains the actions that will drive a raft state machine.
+//!
+//! # Raft Ready
+//!
+//! Every messages or ticks may have side affect. Handling all those side
+//! affect immediately is not efficient. Instead, tikv uses `Ready` to batch up
+//! all the side affects and handle them at once for throughput.
+//!
+//! As raft store is the critical path in the whole system, we avoid most
+//! blocking IO. So a typical processing is divided into two steps:
+//!
+//! - Handle raft ready to process the side affect and send IO tasks to
+//!   background threads
+//! - Receive IO tasks completion and update the raft state machine
+//!
+//! There two steps can be processed concurrently.
+
+mod async_writer;
+
+use engine_traits::{KvEngine, RaftEngine};
+use error_code::ErrorCodeExt;
+use kvproto::raft_serverpb::RaftMessage;
+use protobuf::Message as _;
+use raft::{eraftpb, Ready};
+use raftstore::store::{FetchedLogs, Transport, WriteTask};
+use slog::{debug, error, trace, warn};
+
+pub use self::async_writer::AsyncWriter;
+use crate::{
+    batch::StoreContext,
+    fsm::{PeerFsm, PeerFsmDelegate},
+    raft::{Peer, Storage},
+    PeerTick,
+};
+
+impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> PeerFsmDelegate<'a, EK, ER, T> {
+    /// Raft relies on periodic ticks to keep the state machine sync with other
+    /// peers.
+    pub fn on_raft_tick(&mut self) {
+        if self.fsm.peer_mut().tick() {
+            self.fsm.peer_mut().set_has_ready();
+        }
+        self.schedule_raft_tick();
+    }
+
+    pub fn schedule_raft_tick(&mut self) {
```

Comment:
i think this function is a bit redundant.  Just call self.schedule_tick(PeerTick::Raft) is straightforward enough.